src/index.js.flow

```diff
@@ -131,10 +128,8 @@ declare export function graphql<TResult, TProps, TChildProps>(
 ): OperationComponent<TResult, TProps, TChildProps>;
 
 declare export function withApollo<TProps>(
-  component:
-    | StatelessComponent<TProps & ApolloClient>
-    | Class<React$Component<any, TProps & ApolloClient, any>>,
-): Class<React$Component<void, TProps & ApolloClient, void>>;
+  component: React$ComponentType<TProps & ApolloClient>,
```

Comment:
You might need to make the intersection the other order. i.e. `ApolloClient & TProps`. Given how we implement intersections in Flow, order matters 😊

Reply:
Fixed - the test for withApollo is though rather crude, not sure it's correctly implemented